processing/src/main/java/io/druid/query/filter/InDimFilter.java

```diff
@@ -0,0 +1,57 @@
+package io.druid.query.filter;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.metamx.common.StringUtils;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.List;
+
+public class InDimFilter implements DimFilter
+{
+  private final List<String> values;
+  private final String dimension;
+
+  @JsonCreator
+  public InDimFilter(@JsonProperty("dimension") String dimension, @JsonProperty("values") List<String> values)
+  {
+    this.values = (values == null) ? Collections.<String>emptyList() : values;
+    this.dimension = dimension;
+  }
+
+  @JsonProperty
+  public String getDimension()
+  {
+    return dimension;
+  }
+
+  @JsonProperty
+  public List<String> getValues()
+  {
+    return values;
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    byte[] dimensionBytes = StringUtils.toUtf8(dimension);
+    final byte[][] valuesBytes = new byte[values.size()][];
+    int valuesBytesSize = 0;
+    int index = 0;
+    for (String value : values) {
+      valuesBytes[index] = StringUtils.toUtf8(value);
+      valuesBytesSize += valuesBytes[index].length + 1;
+      ++index;
+    }
+
+    ByteBuffer filterCacheKey = ByteBuffer.allocate(1 + dimensionBytes.length + valuesBytesSize)
+                                          .put(DimFilterCacheHelper.IN_CACHE_ID)
+                                          .put(dimensionBytes);
```

Comment:
@b-slim One more 0xff after the dimensionBytes and this looks good to me (or else the dimension name could get confused with the first value).